src/main/java/com/mraof/minestuck/item/armor/JetPackItem.java

```diff
@@ -0,0 +1,217 @@
+		{
+			damageGear(player, stack);
+		}
+		
+		return true;
+	}
+	
+	public static void damageGear(Player player, ItemStack stack)
+	{
+		Item footItem = player.getItemBySlot(EquipmentSlot.CHEST).getItem();
+		
+		if (!player.level().isClientSide && (player.getFallFlyingTicks() + 1) % 20 == 0) {
+			stack.hurtAndBreak(1, player, p -> p.broadcastBreakEvent(EquipmentSlot.CHEST));
+			if (footItem instanceof JetPackItem)
+				player.getItemBySlot(EquipmentSlot.CHEST).hurtAndBreak(1, player, p -> p.broadcastBreakEvent(EquipmentSlot.CHEST));
+		}
+	}
+	
+	public static void explodeAndStop(Player player, Level level)
+	{
+		ItemStack jetpackItemStack = player.getItemBySlot(EquipmentSlot.CHEST);
+		
+		jetpackItemStack.getOrCreateTag().putBoolean("is_boosting", false);
+		level.explode(player,player.getX() + 0.5D, player.getY() + 0.5D, player.getZ() + 0.5D, 3F, Level.ExplosionInteraction.BLOCK);
+	}
+	
+	public boolean isHoldingControllers(LivingEntity entity)
+	{
+		Item thrustController = MSItems.THRUST_CONTROLLER.get();
+		
+		boolean hasController = entity.getItemInHand(InteractionHand.MAIN_HAND).is(thrustController);
+		boolean hasControllerOffhand =  entity.getItemInHand(InteractionHand.OFF_HAND).is(thrustController);
+		if(hasController && hasControllerOffhand)
+		{
+			return true;
+		}
+		return false;
+	}
+	
+	public static void boost(Player player)
+	{
+		Item chestItem = player.getItemBySlot(EquipmentSlot.CHEST).getItem();
+		if (chestItem instanceof JetPackItem)
+		{
+			player.level().addParticle(MSParticleType.EXHAUST.get(), player.getX(), player.getY() + 0.8, player.getZ() - 0.3, 0, 0, 0);
+			player.level().addParticle(MSParticleType.EXHAUST.get(), player.getX(), player.getY() + 0.8, player.getZ() + 0.3, 0, 0, 0);
+			player.level().playSound(player, player.blockPosition(), MSSoundEvents.JETPACK_THRUST.get(), SoundSource.PLAYERS, 1, 0);
+			
+			Vec3 look = player.getLookAngle();
+			Vec3 movement = player.getDeltaMovement();
+			player.setDeltaMovement(movement.add(
+					look.x * 0.2D + (look.x * 2D - movement.x) * 0.3D,
+					look.y * 0.2D + (look.y * 2D - movement.y) * 0.3D,
+					look.z * 0.2D + (look.z * 2D - movement.z) * 0.3D));
+		}
+	}
+	
+	public void animationStateChange(ItemStack pStack, Level pLevel, Entity pEntity)
+	{
+		if(!isHoldingControllers((LivingEntity) pEntity) && pLevel instanceof ServerLevel serverLevel)
+		{
+			triggerAnim(pEntity, GeoItem.getOrAssignId(pStack, serverLevel), "jetpack_controller", "idle");
+		}
+		if(isHoldingControllers((LivingEntity) pEntity) && pLevel instanceof ServerLevel serverLevel)
+		{
+			triggerAnim(pEntity, GeoItem.getOrAssignId(pStack, serverLevel), "jetpack_controller", "primed");
+		}
+		if(isHoldingControllers((LivingEntity) pEntity) && isBoostingTagTrue(pStack) && pLevel instanceof ServerLevel serverLevel)
+		{
+			triggerAnim(pEntity, GeoItem.getOrAssignId(pStack, serverLevel), "jetpack_controller", "flight");
+		}
```

Comment:
As you need the instanceof cast in all three blocks, I suggest moving it back to only do it once. Potentially as an early return, or to before calling this function (changing the level paremeter to use `ServerLevel` as type).